src/components/application_manager/test/resumption/resume_ctrl_test.cc

```diff
@@ -646,6 +702,16 @@ TEST_F(ResumeCtrlTest, StartResumption_AppWithSubscriptionToWayPoints) {
           GetSavedApplication(kTestPolicyAppId_, kMacAddress_, _))
       .WillByDefault(DoAll(SetArgReferee<2>(saved_app), Return(true)));
   EXPECT_CALL(*mock_app_, set_grammar_id(kTestGrammarId_));
+
+  smart_objects::SmartObjectSPtr subscribe_waypoints_msg;
+  EXPECT_CALL(*application_manager::MockMessageHelper::message_helper_mock(),
+              CreateSubscribeWayPointsMessageToHMI(_))
+      .WillRepeatedly(Return(subscribe_waypoints_msg));
+  std::list<application_manager::AppExtensionPtr> extensions;
+  // It will work only for SDLAppExtension, need to rework this test
```

Comment:
We use the polymorphism here https://github.com/LuxoftSDL/sdl_core/blob/feature/sdl_0190_resumption_data_error_handling/src/components/application_manager/src/resumption/resumption_data_processor.cc#L858 
WayPoints subscriptions are specific for the SDLAppExtension. Since we are not able to create the real SDLAppExtension here, I'm confused. Will be thankful for any ideas

Reply:
Replace inheritance with aggregation almost always works.